### ISSUE TITLE
Theme.of(context).colorScheme.surfaceVariant color is different from FlexSchemeData

### ISSUE THREAD

**Issue by balaji101010** [rydmike/flex_color_scheme#118]
```
const ColorScheme lightColorScheme = ColorScheme.light(
  primary: Color(0xffffcf33),
  primaryContainer: Color(0xfffff3cc),
  onPrimary: Color(0xff191919),
  onPrimaryContainer: Color(0xff997500),
  secondary: Color(0xff191919),
  secondaryContainer: Color(0xffe1e1e1),
  onSecondary: Color(0xffffffff),
  onSecondaryContainer: Color(0xff5d5d5d),
  error: Color(0xffbb1b2e),
  errorContainer: Color(0xfff9d2d7),
  onError: Color(0xffffffff),
  onErrorContainer: Color(0xff590d16),
  background: Color(0xffffffff),
  onBackground: Color(0xff191919),
  surface: Color(0xfffafafa),
  onSurface: Color(0xff191919),
  surfaceVariant: Color(0xffFF0000), // I set red here
  onSurfaceVariant: Color(0xff868686),
  outline: Color(0xffe2e2e2),
);

///
/// Need to define dark color scheme
///
const ColorScheme darkColorScheme = lightColorScheme;

final ThemeData lightTheme = FlexThemeData.light(
  colorScheme: lightColorScheme,
  tooltipsMatchBackground: true,
  useMaterial3: true,
  scaffoldBackground: lightColorScheme.background,
  visualDensity: FlexColorScheme.comfortablePlatformDensity,
  fontFamily: defaultFontFamily,
  textTheme: textTheme.apply(
      bodyColor: lightColorScheme.onBackground,
      displayColor: lightColorScheme.onBackground),
);

final ThemeData darkTheme = FlexThemeData.dark(
  colorScheme: darkColorScheme,
  useMaterial3: false,
  surfaceMode: FlexSurfaceMode.highScaffoldLowSurface,
  visualDensity: FlexColorScheme.comfortablePlatformDensity,
  tooltipsMatchBackground: true,
  scaffoldBackground: darkColorScheme.background,
  fontFamily: defaultFontFamily,
  textTheme: textTheme.apply(
      bodyColor: darkColorScheme.onBackground,
      displayColor: darkColorScheme.onBackground),
);


```

In Screen,

When I use `Theme.of(context).colorScheme.surfaceVariant` the red color is not coming

**Comment by rydmike** [comment]
Thanks for the issue report. Which version of FlexColorScheme was this done with?

**Comment by rydmike** [comment]
I asked about version, because this is a known bug that exists in version 6, but not in version 7.

If I run the tests below they do fail in version 6, but pass in version 7.

```dart
void main() {
  group('FCS8: ISSUE-118-Light WITH FlexColorScheme.toTheme ', () {
    // Tests for issue:
    // https://github.com/rydmike/flex_color_scheme/issues/118
    TestWidgetsFlutterBinding.ensureInitialized();
    test(
        'FCS-ISSUE-118-Light-M2: GIVEN a FlexColorScheme.light with defined '
        'colorScheme, expect toScheme and toTheme to contain the ColorScheme, '
        'provided that no seeding or surface blend is used.', () {
      // Make a full ColorScheme from a SeedColor, and also override
      // surfaceVariant with pure red.
      final ColorScheme schemeLight = ColorScheme.fromSeed(
        brightness: Brightness.light,
        seedColor: const Color(0xffffcf33),
      ).copyWith(
        surfaceVariant: const Color(0xffff0000),
      );
      final FlexColorScheme fcs = FlexColorScheme.light(
        colorScheme: schemeLight,
        tooltipsMatchBackground: true,
        useMaterial3: false,
        scaffoldBackground: schemeLight.background,
        visualDensity: FlexColorScheme.comfortablePlatformDensity,
      );
      final ColorScheme scheme = fcs.toScheme;
      final ThemeData theme = fcs.toTheme;
      expect(
        scheme,
        equals(schemeLight),
      );
      expect(
        theme.colorScheme,
        equals(schemeLight),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(schemeLight.surfaceVariant),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(const Color(0xffff0000)),
      );
    });
    test(
        'FCS-ISSUE-118-Dark-M2: GIVEN a FlexColorScheme.dark with defined '
        'colorScheme, expect toScheme and toTheme to contain the ColorScheme, '
        'provided that no seeding or surface blend is used.', () {
      // Make a full ColorScheme from a SeedColor, and also override
      // surfaceVariant with pure red.
      final ColorScheme schemeDark = ColorScheme.fromSeed(
        brightness: Brightness.dark,
        seedColor: const Color(0xffffcf33),
      ).copyWith(
        surfaceVariant: const Color(0xffff0000),
      );
      final FlexColorScheme fcs = FlexColorScheme.dark(
        colorScheme: schemeDark,
        tooltipsMatchBackground: true,
        useMaterial3: false,
        scaffoldBackground: schemeDark.background,
        visualDensity: FlexColorScheme.comfortablePlatformDensity,
      );
      final ColorScheme scheme = fcs.toScheme;
      final ThemeData theme = fcs.toTheme;
      expect(
        scheme,
        equals(schemeDark),
      );
      expect(
        theme.colorScheme,
        equals(schemeDark),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(schemeDark.surfaceVariant),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(const Color(0xffff0000)),
      );
    });
    test(
        'FCS-ISSUE-118-Light-M3: GIVEN a FlexColorScheme.light with defined '
        'colorScheme, expect toScheme and toTheme to contain the ColorScheme, '
        'provided that no seeding or surface blend is used.', () {
      // Make a full ColorScheme from a SeedColor, and also override
      // surfaceVariant with pure red.
      final ColorScheme schemeLight = ColorScheme.fromSeed(
        brightness: Brightness.light,
        seedColor: const Color(0xffffcf33),
      ).copyWith(
        surfaceVariant: const Color(0xffff0000),
      );
      final FlexColorScheme fcs = FlexColorScheme.light(
        colorScheme: schemeLight,
        tooltipsMatchBackground: true,
        useMaterial3: true,
        scaffoldBackground: schemeLight.background,
        visualDensity: FlexColorScheme.comfortablePlatformDensity,
      );
      final ColorScheme scheme = fcs.toScheme;
      final ThemeData theme = fcs.toTheme;
      expect(
        scheme,
        equals(schemeLight),
      );
      expect(
        theme.colorScheme,
        equals(schemeLight),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(schemeLight.surfaceVariant),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(const Color(0xffff0000)),
      );
    });
    test(
        'FCS-ISSUE-118-Dark-M3: GIVEN a FlexColorScheme.dark with defined '
        'colorScheme, expect toScheme and toTheme to contain the ColorScheme, '
        'provided that no seeding or surface blend is used.', () {
      // Make a full ColorScheme from a SeedColor, and also override
      // surfaceVariant with pure red.
      final ColorScheme schemeDark = ColorScheme.fromSeed(
        brightness: Brightness.dark,
        seedColor: const Color(0xffffcf33),
      ).copyWith(
        surfaceVariant: const Color(0xffff0000),
      );
      final FlexColorScheme fcs = FlexColorScheme.dark(
        colorScheme: schemeDark,
        tooltipsMatchBackground: true,
        useMaterial3: true,
        scaffoldBackground: schemeDark.background,
        visualDensity: FlexColorScheme.comfortablePlatformDensity,
      );
      final ColorScheme scheme = fcs.toScheme;
      final ThemeData theme = fcs.toTheme.copyWith(
          colorScheme: schemeDark.copyWith(
        surfaceVariant: schemeDark.surfaceVariant,
      ));
      expect(
        scheme,
        equals(schemeDark),
      );
      expect(
        theme.colorScheme,
        equals(schemeDark),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(schemeDark.surfaceVariant),
      );
      expect(
        theme.colorScheme.surfaceVariant,
        equals(const Color(0xffff0000)),
      );
    });
  });
}
```

## Fixed in v7

I noticed this issue as well when developing version 7. So this issue has been fixed in version **7.0.0-dev.3** and will also be included in **stable 7.0.0** when that is released. The current **7.0.0-dev.3** is final v7 beta. API wise v7.0.0 stable will when released be identical. Currently it is only undergoing final testing, it will not get any more features or API changes.

### Back patch to v6?

Using `FlexColorScheme` with fully custom provided `ColorScheme` is a rather rare use case. While it is intended to work, I'm not sure there is any major benefit in fixing it anymore. Especially since it is fixed in version 7. If it is really critical to get a fix in v6, I can look into it later.

Some alternatives below:

### Use v7

If you can upgrade, I recommend trying **version 7.0.0-dev.3**. Which is actually much better, especially if you are using Flutter 3.7.

Flutter 3.7 was a really big release when it comes to theming and Material 3. FlexColorScheme v7 fully supports it, whereas version 6, does not, since it was released before Flutter 3.7. Version 6 works with Flutter 3.7 too, but not always in an ideal way design wise, especially not if using Material 3.

### Fix using `copyWith`

If you cannot upgrade, and need the fix in v6 right now, you can also correct the `ColorScheme` manually by using `copyWith` on the `ThemeData` returned by `FlexThemeData`.

The tests above actually show an example of doing it with the single `surfaceVariant` color.

Since you fully define the `ColorScheme` externally, and don't change the outcome by using any seeding or surface blends in `FlexColorScheme`, you can actually override the entire `ColorScheme` with your externally defined `ColorScheme`, which is a bit sorter than the example above.

It would look like:

```dart
final ThemeData lightTheme = FlexThemeData.light(
  colorScheme: lightColorScheme,
  tooltipsMatchBackground: true,
  useMaterial3: true,
  scaffoldBackground: lightColorScheme.background,
  visualDensity: FlexColorScheme.comfortablePlatformDensity,
  fontFamily: defaultFontFamily,
  textTheme: textTheme.apply(
      bodyColor: lightColorScheme.onBackground,
      displayColor: lightColorScheme.onBackground),
).copyWith(colorScheme: lightColorScheme);
```

**Comment by balaji101010** [comment]
Sorry for the delay. I am using 6.1.2
